src/gpuarray_array.c

```diff
@@ -1117,8 +1121,7 @@ int GpuArray_is_c_contiguous(const GpuArray *a) {
   int i;
 
   for (i = a->nd - 1; i >= 0; i--) {
-    if (a->dimensions[i] != 1 && a->strides[i] != size)
-      return 0;
+    if (a->strides[i] != size) return 0;
```

Comment:
Just to be clear, the problem is that the case you give is only f
contiguous and not c contiguous. But currently it
get detected as both.

My proposed could be reformulated as (but it don't work for your case)

if ((a->dimensions[i] != 1 && a->strides[i] != size) || (a->strides[i] %
size == 0)) return 0;

Instead of trying to find what to do, can you check what the master of
numpy does?


On Thu, Jul 21, 2016 at 10:25 AM, abergeron <notifications@github.com>
wrote:

> In src/gpuarray_array.c
> <https://github.com/Theano/libgpuarray/pull/226#discussion_r71714821>:
>
> > @@ -1117,8 +1121,7 @@ int GpuArray_is_c_contiguous(const GpuArray *a) {
> >    int i;
> >
> >    for (i = a->nd - 1; i >= 0; i--) {
> > -    if (a->dimensions[i] != 1 && a->strides[i] != size)
> > -      return 0;
> > +    if (a->strides[i] != size) return 0;
>
> If I understand the intent of the code shape = (1, 3, 1, 1) and strides =
> (4, 4, 12, 12) would pass as C-contiguous when it is obviously not. This
> is what I'm trying to fix.
>
> Otherwise I don't get it since the check is written in a confusing way
> that completely changes the way it was done before.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Theano/libgpuarray/pull/226/files/aa9cdb1d5d6fe7d94a0036b67fcffa45717f7051#r71714821>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AALC-3ookBwSMQqJayNtdyAckM_WElRRks5qX4FAgaJpZM4JRLDV>
> .
>